HT6_3.py

```diff
@@ -0,0 +1,228 @@
+                return True
+            else:
+                print("File format is empty.\nPlease check folder templates_for_adding_publication for correct format.")
+        else:
+            print("File with this name is not exist. Try again.")
+
+    def get_file_name(self):
+        while True:
+            fname = f'{input("Please give name of txt file: ")}.{self.publication_method}'
+            if self.__is_valid_file_name(fname):
+                print(f"You selected file is {fname}")
+                return fname
+
+    def read_from_txt(self):
+        with open(f"{self.file_name}") as file:
+            input_list = file.readlines()
+            input_list = [row.replace('\n','') for row in input_list]
+        while True:
+            if self.__is_valid_txt_format(input_list):
+                return input_list
+
+    def __is_valid_txt_format(self, input_list):
+        if "Publication_Type: " in input_list[0] and "Body: " in input_list[1]:
+            return True
+        else:
+            print("File format is invalid.\nPlease check folder templates_for_adding_publication for correct format.")
+            self.get_file_name()
+
+    def get_publication_type(self):
+        if "News" in self.input_list[0]:
+            print("Your publication type is News")
+            return News(publish_datetime=datetime.datetime.now(), publication_type="News")
+        elif "Private Ad" in self.input_list[0]:
+            print("Your publication type is Ad")
+            return Ad(publication_type="Ad")
+        elif "Joke" in self.input_list[0]:
+            print("Your publication type is Joke")
+            return Joke(publication_type="Joke")
+        else:
+            print(f"Invalid file format. Correct format is:\n\tPublication_Type: \n\tBody: \n\tCity: or Expiration_Day:")
+
+    def get_publication_text(self):
+        body = self.input_list[1][self.input_list[1].index(': ') + 2:]
+        print(f"Publication text is:\n\t{body}")
+        return body
+
+    def get_publication_specific_info(self, publication: Publication) -> str:
+        if publication.publication_type == "News":
+            publication.location = self.input_list[2][self.input_list[2].index(': ') + 2:]
+            print(f"News location is {publication.location}")
+            return publication.location
+        elif publication.publication_type == "Ad":
+            exp_date = self.input_list[2][self.input_list[2].index(': ') + 2:]
+            exp_date = datetime.datetime.strptime(exp_date, "%Y-%m-%d").date()
+            date_diff = exp_date - datetime.date.today()
+            if date_diff.days < 0:
+                print('Your advertising is expired. It will not be published.')
+                exit()
+            publication.expiration_date = f'Actual until: {exp_date}, {date_diff.days} days left'
+        elif publication.publication_type == "Joke":
+            publication.funny_meter = publication.generate_random_funny_meter()

```

Comment:
The same recommendation here:
if you are referring to the functions of the child classes in the method, it is better to refer to typing.Union[News,Ad,Joke,OtherChildClass] in the signature